modules/component/src/core/render-scheduler.ts

```diff
@@ -30,3 +35,7 @@ export function createRenderScheduler(
 function hasZone(z: NgZone): boolean {
   return z instanceof NgZone;
 }
+
+function getCdRefContext(cdRef: ChangeDetectorRef): object {
+  return (cdRef as unknown as { context: object }).context;
```

Comment:
Does `cdRef` always have a `context` here?

Reply:
When the `ChangeDetectorRef` instance comes from a pipe or directive as in our case, the context will be always defined (it will be a host component).